lib/media/track/remotemediatrack.js

```diff
@@ -137,6 +151,35 @@ function mixinRemoteMediaTrack(AudioOrVideoTrack) {
   };
 }
 
+function playIfPausedWhileInBackground(audioTrack) {
+  const { _log: log } = audioTrack;
+  let wasVisible = document.visibilityState === 'visible';
+  let elementsPlaying = [];
+
+  document.addEventListener('visibilitychange', () => {
```

Comment:
@makarandp0 I think this will result in event listener leaks when this RemoteMediaTrack is unsubscribed. I would rather listen to the `pause` event on the elements:
```js
function playIfPausedWhileInBackground(remoteMediaTrack) {
  remoteMediaTrack._attachments.forEach(el => {
    el.addEventListener('pause', () => {
      if (document.visibilityState === 'hidden') {
        // This ensures no event listener leaks as it is removed as soon as it is fired once.
        waitForEvent(document, 'visibilitychange').then(() => el.play());
      }
    });
  });
}
```

Reply:
@makarandp0 Also, we need to make sure we detach the pause event listener when the element is `detach()`ed.

Reply:
Thanks @manjeshbhargav for catching the listener leak. One problem I see with listening on individual elements is they are added/removed dynamically, requiring that we hook/unhook on them individually as they are attach/detach'ed. It also means maintaining a map of el => listener so that we can remove those listeners when the elements are detached.

What do you think about keep listening on document visibility, but unhook listener when there are no elements attached. I have updated code to unhook the listener now.